examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,579 @@
+    corrected_std = np.sqrt(corrected_var)
+    return corrected_std
+
+
+def compute_corrected_ttest(differences, df, n_train, n_test):
+    """Computes right-tailed paired t-test with corrected variance.
+
+    Parameters
+    ----------
+    differences : array-like of shape (n_samples, 1)
+        Vector containing the differences in the score metrics of two models.
+    df : int
+        Degrees of freedom.
+    n_train : int
+        Number of samples in the training set.
+    n_test : int
+        Number of samples in the testing set.
+
+    Returns
+    -------
+    t_stat : float
+        Variance-corrected t-statistic.
+    p_val : float
+        Variance-corrected p-value.
+    """
+    mean = np.mean(differences)
+    std = corrected_std(differences, n_train, n_test)
+    t_stat = mean / std
+    p_val = t.sf(np.abs(t_stat), df)  # right-tailed t-test
+    return t_stat, p_val
+
+
+# %%
+model_1_scores = model_scores.iloc[0].values  # scores of the best model
+model_2_scores = model_scores.iloc[1].values  # scores of the second-best model
+
+differences = model_1_scores - model_2_scores
+
+n = differences.shape[0]  # number of test sets
+df = n - 1
+n_train = len(list(cv.split(X, y))[0][0])
+n_test = len(list(cv.split(X, y))[0][1])
+
+t_stat, p_val = compute_corrected_ttest(differences, df, n_train, n_test)
+print(f"Corrected t-value: {t_stat:.3f}\n"
+      f"Corrected p-value: {p_val:.3f}")
+
+# %%
+# We can compare the corrected t- and p-values with the uncorrected ones:
+
+t_stat_uncorrected = (
+    np.mean(differences) / np.sqrt(np.var(differences, ddof=1) / n)
+)
+p_val_uncorrected = t.sf(np.abs(t_stat_uncorrected), df)
+
+print(f"Uncorrected t-value: {t_stat_uncorrected:.3f}\n"
+      f"Uncorrected p-value: {p_val_uncorrected:.3f}")
+
+# %%
+# Using the conventional significance alpha level at `p=0.05`, we observe that
+# the uncorrected t-test concludes that the first model is significantly better
+# than the second.
+#
+# With the corrected approach, in contrast, we fail to detect this difference.
+#
+# In the latter case, however, the frequentist approach does not let us
+# conclude that the first and second model have an equivalent performance. If
+# we wanted to make this assertion we need to use a Bayesian approach.
+
+# %%
+# Comparing two models: Bayesian approach
+# ---------------------------------------
+# We can use Bayesian estimation to calculate the probability that the first
+# model is better than the second. Bayesian estimation will output a
+# distribution that specifies the probability of each parameter value, which in
+# this example is the mean of the differences in the performance of two models.
+#
+# To obtain the posterior distribution we need to define a prior that models
+# our beliefs of how the means are distributed before looking at the data,
+# and multiply it by a likelihood function that computes how likely our
+# observed mean differences are, given the values that the mean of differences
```

Comment:
about "observed mean differences":

I find this a bit strange because as far as I understand, we don't really observe multiple "mean differences"; we observe multiple multiple differences, but this account for just *one* empirical mean difference?

Reply:
In this case it should be the likelihood of the observed differences

Reply:
here the likelihood should be how likely our differences are, given the (...)